### ISSUE TITLE
DBSCAN hybrid mode with the cluster_selection_epsilon does not support soft clustering on out of sample data

### ISSUE THREAD

**Issue by Dicksonchin93** [scikit-learn-contrib/hdbscan#526]
DBSCAN hybrid mode with the `cluster_selection_epsilon` parameter set to a value more than 0 does not support soft clustering on out of sample data

We don't utilise `cluster_selection_epsilon` anywhere in the `membership_vector` method in  https://github.com/scikit-learn-contrib/hdbscan/blob/4c432505f4a92884a64a77159664f041a583fbec/hdbscan/prediction.py#L518

The suggested part to add support for that is to add the same logic during fitting with `cluster_selection_epsilon` is in the `select_clusters` method used here https://github.com/scikit-learn-contrib/hdbscan/blob/4c432505f4a92884a64a77159664f041a583fbec/hdbscan/prediction.py#L550 

https://github.com/scikit-learn-contrib/hdbscan/blob/4c432505f4a92884a64a77159664f041a583fbec/hdbscan/plots.py#L234

**Comment by lmcinnes** [comment]
Yes, I believe this is an interaction of features that is not going to manage to work. Sorry.
